In [ ]:
# Mengimpor library yang dibutuhkan
import scipy.stats as stats

In [ ]:
# Mempersiapkan data
data = (100, 150, 50, 100, 130, 120, 100, 110, 75, 65, 150, 120, 50, 100, 100, 140, 90, 150, 50, 90)

## Menentukan H0, H1 dan alpha

H0: Mean = 100

H1: Mean > 100

alpha = 0.05

In [ ]:
# Melakukan one sample t-test
stats.ttest_1samp(a=data, popmean=100)

Ttest_1sampResult(statistic=0.2731657711666385, pvalue=0.787672894169515)

T-statistic bernilai 0.2731657711666385

T-statistic = 0.27 (pembulatan)

## One tailed test

In [ ]:
# Jika menggunakan statistik t, kita harus mencari critical regionnya terlebih dahulu:
# Jika H1: mean > 100 (sisi kanan)
# argumen pertama adalah 1-alpha = 1-0.05
# argumen kedua adalah degree of freedom = N-1 = 20-1 = 19
stats.t.ppf(1-0.05, 19)

1.729132811521367

Critical Region: t > 1.73 (pembulatan)

Karena T-statistic tidak berada di Critical Region, maka kita gagal menolak H0

**Kesimpulan: Rata-rata transaksi setelah campaign = 100. Campaign itu tidak berhasil.**